src/gui/state.rs

```diff
@@ -259,6 +264,7 @@ impl UiState {
 
     /// Pop a new UI rendering position from the stack.
     #[inline]
+    #[allow(clippy::similar_names)]
```

Comment:
I didn't know what `pcursor` stood for, so I decided to disable a lint here and get some snacks (srsly tho, what `pcursor` stands for?)

Reply:
cursor is for the current UI rendering position, pcursor is for the previous rendering position. They could likely benefit from a clearer rename - but I was staying consistent with mouse and pmouse used in `PixState`.

In general I've not been as diligent on the naming and documentation of internals, though it would be nice to clean up some of these areas to help future contributors